packages/core/lib/dlc/finance/OptionInfo.ts

```diff
@@ -65,10 +71,26 @@ export function getOptionInfoFromContractInfo(
   if (payoutCurvePiece.b !== BigInt(0) || payoutCurvePiece.c !== BigInt(0))
     throw Error('b and c HyperbolaPayoutCurvePiece values must be 0');
 
+  const curve = HyperbolaPayoutCurve.fromPayoutCurvePiece(payoutCurvePiece);
+  const maxOutcome = BigInt(
+    new BN(eventDescriptor.base)
+      .pow(eventDescriptor.nbDigits)
+      .minus(1)
+      .toString(10),
+  );
+  const isAscending = curve
+    .getPayout(maxOutcome)
+    .gt(Number(payoutFunction.endpointPayout0));
```

Comment:
So for now it only checks if it's ascending or not to determine if it's a put or call? I suspect there are lots of edge cases where this is not the case. Any reason for not running it through `buildPayoutFunction` again?

Reply:
More secure differentiation is a separate task in Linear: https://linear.app/atomicfinance/issue/ATOM-216/implement-a-more-secure-way-of-differentiating-between-a-put-and-call

Currently in progress is rebuilding the curve and implementing and performing an `isEquals` check on the two curves.

Reply:
Added sanity check